ComponentKit/LayoutComponents/CKStackUnpositionedLayout.mm

```diff
@@ -256,30 +255,34 @@ static void flexChildrenAlongStackDimension(std::vector<CKStackUnpositionedItem>
 {
   const CGFloat stackDimensionSum = computeStackDimensionSum(items, style);
   const CGFloat violation = computeViolation(stackDimensionSum, style, sizeRange);
-
-  // We count the number of children which are flexible in the direction of the violation
-  std::function<BOOL(const CKStackUnpositionedItem &)> isFlex = isFlexibleInViolationDirection(violation);
-  const NSUInteger flexibleChildren = std::count_if(items.begin(), items.end(), isFlex);
-  if (flexibleChildren == 0) {
-    // If optimized flexing was used then we have to clean up the unsized children, and lay them out at zero size
+  // First compute the sum of all flex-grow/flex-shrink values, or flex factors, of children in the direction of the violation.
+  std::function<BOOL(const CKStackUnpositionedItem &)> flexFactor = flexFactorInViolationDirection(violation);
+  const NSInteger flexFactorSum = std::accumulate(items.begin(), items.end(), 0, [&](NSInteger x, const CKStackUnpositionedItem &l) {
+    return x + flexFactor(l);
+  });
+  if (flexFactorSum == 0) {
+    // If optimized flexing was used then we have to clean up the unsized children, and lay them out at zero size.
     if (useOptimizedFlexing) {
       layoutFlexibleChildrenAtZeroSize(items, style, sizeRange, size);
     }
     return;
   }
-
-  // Each flexible child along the direction of the violation is expanded or contracted equally
-  const CGFloat violationPerFlexChild = floorf(violation / flexibleChildren);
+  // Next distribute the flex factor sum evenly across the violation. Flexible children will be given proportional (based
+  // on their flex factors) increments of this value to flex appropriately.
+  const CGFloat violationPerFlexibleChild = floorf(violation / flexFactorSum);
   // If the floor operation above left a remainder we may have a remainder after deducting the adjustments from all the
   // contributions of the flexible children.
-  const CGFloat violationRemainder = violation - (violationPerFlexChild * flexibleChildren);
-
+  const CGFloat violationRemainder = violation - (violationPerFlexibleChild * flexFactorSum);
   BOOL isFirstFlex = YES;
   for (CKStackUnpositionedItem &item : items) {
-    if (isFlex(item)) {
+    const NSInteger currentFlexFactor = flexFactor(item);
+    // Children are considered inflexible if they their flex factor is 0 (i.e. both flex-grow and flex-shrink are 0).
+    if (currentFlexFactor != 0) {
       const CGFloat originalStackSize = stackDimension(style.direction, item.layout.size);
-      // The first flexible child is given the additional violation remainder
-      const CGFloat flexedStackSize = originalStackSize + violationPerFlexChild + (isFirstFlex ? violationRemainder : 0);
+      // Now adjust each flexible child by applying the remaining violation based on the flex factor.
+      const CGFloat flexAdjustment = violationPerFlexibleChild * currentFlexFactor;
```

Comment:
I'm not sure it's this simple. I think the scaled flex factor shoudl be used here instead?